### ISSUE TITLE
Error in migration when updating enum

### ISSUE THREAD

**Issue by PauloPraf** [prisma/prisma#15027]
Hi Prisma Team! Prisma Migrate just crashed. 

## Command

`migrate dev`

## Versions
      
| Name        | Version            |
|-------------|--------------------|
| Platform    | windows            | 
| Node        | v16.15.0           | 
| Prisma CLI  | 4.2.1              | 
| Engine      | 2920a97877e12e055c1333079b8d19cee7f33826| 

## Error
```
Error: Error in migration engine.
Reason: [migration-engine\connectors\sql-migration-connector\src\sql_renderer\postgres_renderer.rs:908:22] We should only be setting a changed default if there was one on the previous schema and in the next with the same enum.

Please create an issue with your `schema.prisma` at
https://github.com/prisma/prisma/issues/new

```

**Comment by danstarns** [comment]
Hey @PauloPraf, 

Please could you share with us your Prisma schema?

**Comment by OscarMulder** [comment]
I had this exact same error with the following schema change. 

Before:
```
model User {
  id            String    @id @default(cuid())
  name          String?
  email         String?   @unique
  emailVerified DateTime?
  image         String?
  accounts      Account[]
  sessions      Session[]
  roles         Role[]    @default([STUDENT])
}

enum Role {
  STUDENT
  STAFF
  MONITORINGMANAGER
  ADMIN
}
```
Changed only the enum:
```
enum Role {
  STUDENT
  STAFF
  STUDENTAFFAIRS
  ADMIN
}
```

I got the following warning beforehand:
```
⚠️  Warnings for the current datasource:

  • The values [MONITORINGMANAGER] on the enum `Role` will be removed. If these variants are still used in the database, this will fail.

✔ Are you sure you want create and apply this migration? … yes
```
However there are no records in my database with the value `MONITORINGMANAGER`.

**Comment by OscarMulder** [comment]
After removing the `@default([STUDENT])` the migration was successfully applied.

**Comment by janpio** [comment]
Was this also about changing an enum for you @PauloPrafß

**Comment by PauloPraf** [comment]
> Was this also about changing an enum for you @PauloPraf?

Yes, it was the same situation as described above.

**Comment by jkomyno** [comment]
Another report: https://prisma-errors.netlify.app/report/14266

**Comment by Jolg42** [comment]
Hi @PauloPraf 👋🏼 
It looks like this was fixed in 4.7.0, could you try upgrading to that version or a more recent one?

PR: https://github.com/prisma/prisma-engines/pull/3379

**Comment by Jolg42** [comment]
@PauloPraf I'll close this as it should be fixed already, let us know if it's indeed the case or if we should reopen this.
